version: '3.1'
services:
  database:
    image: postgres:12.1
    restart: always
    container_name: surprise-postgres-db
    volumes:
      - ./postgres/init.sql:/docker-entrypoint-initdb.d/init.sql
      - postgres:/var/lib/postgresql/data/
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - 5432:5432
  etl:
    build: ./django-etl
    container_name: surprise-etl
    restart: always
    command: python ./django-etl/src/manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
      - ./script:/script
      - ${INITIAL_DATA}:/code/django-etl/initdata
      - ${NEO4j_IMPORT}:/code/django-etl/import
    ports:
      - 8000:8000

volumes:
  postgres:
